### PR TITLE
added/updated tests for comparator set details section

### PR DESCRIPTION
**Pull request by Faizi52996**
### Context
Added tests for comparator set details component for compare your costs & workforce benchmark. 
[AB#199626](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/199626) ([AB#200391](https://dfe-ssp.visualstudio.com/a14e55df-4fbf-4a2f-a11d-22b187178343/_workitems/edit/200391))

### Change proposed in this pull request
Updated existing step and added one for tests for each of the page.

### Guidance to review 
Include any useful information needed to review this change. Include any dependencies that are required for this change. 

### Checklist
- [x] Work items have been linked [(use AB#)](https://learn.microsoft.com/en-us/azure/devops/boards/github/link-to-from-github?view=azure-devops#use-ab-to-link-from-github-to-azure-boards-work-items)
- [x] Your code builds clean without any errors or warnings
- [x] You have run all unit/integration tests and they pass
- [x] Your branch has been rebased onto main
- [x] You have tested by running locally

### CODE REVIEW

**Review by jrabbott — CHANGES_REQUESTED** [review]
Variable names are too verbose.  The shorter the better, and avoid junk words.

Formatting on BenchmarkWorkforce.feature isn't correct.

File: web/tests/Web.E2ETests/Pages/School/BenchmarkWorkforcePage.cs [hunk 1]
```diff
@@ -36,13 +36,24 @@ public class BenchmarkWorkforcePage(IPage page)
...
+    private ILocator HowWeChooseSimilarSchoolsDetailsSection => page.Locator(Selectors.GovDetailsText);
+
     public async Task IsDisplayed()
     {
         await PageH1Heading.ShouldBeVisible();
         await Breadcrumbs.ShouldBeVisible();
         await ChangeSchoolLink.ShouldBeVisible();
         await ViewAsTableRadio.ShouldBeVisible().ShouldBeChecked(false);
         await ViewAsChartRadio.ShouldBeVisible().ShouldBeChecked();
+        await HowWeChooseSimilarSchoolDetailsBtn.ShouldBeVisible();
```
Comment (jrabbott) [inline web/tests/Web.E2ETests/Pages/School/BenchmarkWorkforcePage.cs:54]: This isn't a button, and therefore shouldn't be named as such

Comment (Faizi52996) [reply web/tests/Web.E2ETests/Pages/School/BenchmarkWorkforcePage.cs]: updated it now.

File: web/tests/Web.E2ETests/Pages/School/CompareYourCostsPage.cs [hunk 2]
```diff
@@ -149,6 +157,17 @@ public async Task IsSectionVisible(ComparisonChartNames chartName, bool visibili
         await IsSectionContentVisible(chartName, visibility, chartMode);
     }
 
+    public async Task ClickHowWeChooseSimilarSchoolsBtn()
```
Comment (jrabbott) [inline web/tests/Web.E2ETests/Pages/School/CompareYourCostsPage.cs:160]: Please rename the method - overlay verbose.

File: web/tests/Web.E2ETests/Pages/School/CompareYourCostsPage.cs [hunk 3]
```diff
@@ -29,6 +29,11 @@ public class CompareYourCostsPage(IPage page)
 
     private ILocator SaveAsImageButtons =>
         page.Locator(Selectors.Button, new PageLocatorOptions { HasText = "Save as image" });
+    private ILocator ComparatorSetDetails =>
+        page.Locator(Selectors.GovDetailsSummaryText, new PageLocatorOptions { HasText = "How we choose similar schools" });
+    private ILocator ViewChangeComparatorSetLink => page.Locator(Selectors.GovLink,
```
Comment (jrabbott) [inline web/tests/Web.E2ETests/Pages/School/CompareYourCostsPage.cs:34]: Please rename variable - overlay verbose. "ViewChange" doesn't add any additional value.

File: web/tests/Web.E2ETests/Pages/School/BenchmarkWorkforcePage.cs [hunk 4]
```diff
@@ -133,6 +144,17 @@ public async Task HasDimensionValuesForChart(WorkforceChartNames chartName, stri
         Assert.Equal(expected, actual);
     }
 
+    public async Task ClickHowWeChooseSimilarSchoolsBtn()
```
Comment (jrabbott) [inline web/tests/Web.E2ETests/Pages/School/BenchmarkWorkforcePage.cs:147]: Please rename method - overlay verbose, meaning the code is difficult to read.

File: web/tests/Web.E2ETests/Pages/School/BenchmarkWorkforcePage.cs [hunk 5]
```diff
@@ -36,13 +36,24 @@ public class BenchmarkWorkforcePage(IPage page)
     private ILocator SaveAsImageButtons =>
         page.Locator(Selectors.Button, new PageLocatorOptions { HasText = "Save as image" });
 
+    private ILocator ComparatorSetDetails =>
+        page.Locator(Selectors.GovDetailsSummaryText, new PageLocatorOptions { HasText = "How we choose similar schools" });
+
+    private ILocator ViewChangeComparatorSetLink => page.Locator(Selectors.GovLink,
```
Comment (jrabbott) [inline web/tests/Web.E2ETests/Pages/School/BenchmarkWorkforcePage.cs:42]: Please rename variable - overlay verbose. "ViewChange" doesn't add any additional value and makes the code harder to read.

File: web/tests/Web.E2ETests/Pages/School/CompareYourCostsPage.cs (hunk 3 above)
Comment (Faizi52996) [reply web/tests/Web.E2ETests/Pages/School/CompareYourCostsPage.cs]: removed ViewChange

File: web/tests/Web.E2ETests/Steps/School/CompareYourCostsSteps.cs [hunk 6]
```diff
@@ -130,6 +130,20 @@ public async Task ThenTheSectionIsHidden(string chartName)
         await _comparisonPage.IsSectionVisible(ChartNameFromFriendlyName(chartName), false, "Show", "chart");
     }
 
+    [When("I click on how we choose similar schools")]
+    public async Task ClickComparatorSetDetails()
```
Comment (jrabbott) [inline web/tests/Web.E2ETests/Steps/School/CompareYourCostsSteps.cs:134]: This should be `WhenIClickOnHowWeChooseSimilarSchools` to follow SpecFlow naming convention for step definitions

Comment (Faizi52996) [reply web/tests/Web.E2ETests/Steps/School/CompareYourCostsSteps.cs]: ahh yes good catch. fixed it now.

**Review by jrabbott — APPROVED** [review]

**Pull request merged**